### ISSUE TITLE
defstruct creates an unnecessary "Unresolved Symbol"

### ISSUE THREAD

**Issue by kasajian** [clj-kondo/clj-kondo#1894]
**version**
Nothing happens when I typed: `clj-kondo --version` so I can't tell which version 

I use Calva in VS-Code and somehow this is working.  But I can reproduce the problem in: https://clj-kondo.michielborkent.nl/

**macro usage**

I don't think so.

**platform**

macOS and Windows

**editor**

vscode

**problem**

This code:

         (defstruct Customer :Name :Phone)

I get this error: `Unresolved symbo: Customer`

![image](https://user-images.githubusercontent.com/2384797/205108361-e822a7a1-6ae9-47f2-9748-c42bbc7b63ef.png)

**repro**

See "This code:"

**config**

I don't have a config.

**expected behavior**

Be able to use `defstruct` without a warning

**Comment by borkdude** [comment]
I agree that clj-kondo should support this, but on the other hand, defstruct is hardly ever used by anyone since `deftype` and  `defrecord` came into existence.

You can work around this by doing: `(def name (create-struct keys...))` (since that is the expansion).

**Comment by kasajian** [comment]
@borkdude Thanks for the feedback.  I didn't realize that defrecord is preferred of defstruct.   That's good to know.   Wish there was a list of these things like, what is old, what is new, etc.

Like, in C#, I am pretty familiar with what constructs to avoid 'cause I'm been in that ecosystem (for better or for worse) for a long time, so I've seen the history.   Someone new to a language, they just see the spec and the library reference -- and it's hard to tell what are the things that probably wouldn't have gotten in the language if it was conceived today..  what is legacy, and what are the rough edges to avoid.

For instance, I had to do a ton of research to understand why STM is not in ClojureScript.   My initial reaction to STM was that, "Oh, I get it.  That's how you solve the whole immutability problem"   So I walked way thinking STM was an essential component, imagine my surprise in reading the ClojureScript documentation that it's not supported -- with no explanation as to why.   Like, okay, you don't support it -- I'm sure that's the right decision -- but *why* is that the right decision and what should you do instead?   This type of stuff, the tribal knowledge, is all out there but it requires to much digging.   And when you're a novice, anyone who knows more than you seems like an expert, so it's hard to differentiate between someone who is explaining from idiomatic Clojure point of view, or they have a fringe opinion, the latter being more common on the Internet.

By the way, good work on babashka 👍

**Comment by borkdude** [comment]
All good points. Maybe @puredanger is interested in hearing this feedback as well.
